### DISCUSSION TITLE
GPT-4 integration

### DISCUSSION THREAD

**Discussion by nitefood** [nitefood/ai-bash#2] [General]
Since I do not currently have access to ChatGPT+, I'd like anybody who has it (and also has access to GPT4 API) to provide feedback on the `gpt-4` model support I added into the [gpt4-support](https://github.com/nitefood/ai-bash-gpt/tree/gpt4-support) branch.

Direct link to the test version of the script: https://raw.githubusercontent.com/nitefood/ai-bash-gpt/gpt4-support/ai

**Comment by chnateag** [comment]
Hi, saw your post on reddit, here is the output i get after downloading wget https://raw.githubusercontent.com/nitefood/ai-bash-gpt/gpt4-support/ai and just adding my api key:

```
./ai

Enter an empty line to send your query, or  ?  to view previous conversations)

╭────────────────────────────────────────────────╮
╞  YOU  (conversation topic: "new conversation") ╡
╰────────────────────────────────────────────────╯
» help me to check what sites are managed by nginx


╭───────────────╮
╞  ChatGPT-3.5  ╡
╰───────────────╯
As an AI language model, I don't have access to your server or system files. But I can provide you some ways to check it. Here are some possible methods:

1. Check Nginx Configuration Files: You can look into Nginx configuration files (/etc/nginx/nginx.conf or /etc/nginx/conf.d/*.conf) and check which sites are listed in the server blocks.

2. Check Nginx Access Logs: You can also look into access logs (/var/log/nginx/access.log) and see which sites are being accessed.

3. Check Nginx status page: Nginx has a built-in status page (/nginx_status) that provides information about the current connections, requests and server statistics. You can access this page using a web browser or use the following command to check:
curl http://localhost/nginx_status

4. Use Nginx command-line tools: You can use Nginx command-line tools like nginx -T to check the configuration and see which sites are being managed by Nginx.

5. Use a web port scanner tool: You can use a web port scanner tool like Nmap to scan your server's IP address and see which ports are open. If the port 80 or 443 is open, it may indicate that Nginx is managing a website on your server.

These methods can help you to determine which sites are managed by Nginx.

╭───────────────────────────────────────────────────────────────╮
╞  YOU  (conversation topic: "Checking sites managed by Nginx") ╡
╰───────────────────────────────────────────────────────────────╯
»

```

**Reply by nitefood** [reply]
Hey there,

The actual script is the one in the gpt4-support branch, which you can get by running:

`wget https://raw.githubusercontent.com/nitefood/ai-bash-gpt/gpt4-support/ai`

Afterwards, provided you have a working GPT-4 API key, please use the script by querying ChatGPT4 using `ai -4`. The various commands you can use the `-4` switch with will be outlined by `ai -h`.

thanks for taking the time!

Edit: sorry, just realized you had already downloaded the script from the correct branch - all it takes is to use the `-4` switch when using it then :-)

**Comment by chnateag** [comment]
okay sorry didn't saw that arg, here is the output :
```
- API ERROR -
 TYPE  invalid_request_error
 CODE  model_not_found
 MESG  The model: `gpt-4` does not exist

- API ERROR -
 TYPE  invalid_request_error
 CODE  model_not_found
 MESG  The model: `gpt-4` does not exist

```

edit: sorry, i thought i had access to GPT-4 Api with GPTPlus but i'm not so sure anymore
edit2: i see a waiting list, sorry i gave you hope : /

**Reply by nitefood** [reply]
no problem, thanks for trying. In case you manage to get access to the API in the future, I'll be thankful if you take the time to provide feedback!
